lib/api/mailboxes.js

```diff
@@ -539,6 +558,24 @@ module.exports = (db, server, mailboxHandler) => {
                 requestBody: {
                     path: Joi.string()
                         .regex(/\/{2,}|\/$/, { invert: true })
+                        .max(MAX_MAILBOX_NAME_LENGTH * MAX_SUB_MAILBOXES + 127)
+                        .custom((value, helpers) => {
+                            const splittedPathArr = value.split('/');
```

Comment:
The same validation function is used in multiple places. Instead, define it as a single validation function in https://github.com/nodemailer/wildduck/blob/master/lib/schemas.js and import it.